### PR TITLE
Add oauth to tests

### PR DESCRIPTION
**Pull request by flacatus**
Signed-off-by: Flavius Lacatusu <flacatus@redhat.com>

### What does this PR do?
This PR cover all our openshift tests with OAUTH.
Reff issue: https://github.com/eclipse/che/issues/17939
### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [Custom resource definition file is up to date](https://github.com/eclipse/che-operator/blob/master/README.md#updating-custom-resource-definition-file)
- [ ] [Nightly OLM bundle is up to date](https://github.com/eclipse/che-operator/blob/master/README.md#update-nightly-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by flacatus** [comment]
NOT ready for review yet

File: .github/bin/oauth-provision.sh [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+#
+
+# exit immediately when a command fails
+set -e
+# only exit with zero if all commands of the pipeline exit successfully
+set -o pipefail
+# error on unset variables
+set -u
+
+export CHE_EXPOSURE_STRATEGY="single-host"
```
[the remark this answers is not in the document]
Comment (flacatus) [reply .github/bin/oauth-provision.sh]: My mistake

File: .ci/cico_updates_openshift.sh [hunk 2]
```diff
@@ -15,17 +15,25 @@ set -x
 
 export OPERATOR_REPO=$(dirname $(dirname $(readlink -f "$0")));
 source "${OPERATOR_REPO}"/.github/bin/common.sh
+source "${OPERATOR_REPO}"/.github/bin/oauth-provision.sh
 
 #Stop execution on any error
 trap "catchFinish" EXIT SIGINT
 
+overrideDefaults() {
+  export OAUTH="true"
```
Comment (tolusha) [inline .ci/cico_updates_openshift.sh:24]: Should be removed ?

File: .github/bin/oauth-provision.sh [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    IDP_HOST="https://"$(oc get route keycloak -n eclipse-che -o=jsonpath='{.spec.host}')
+  fi
+
+  # Get the oauth client from Eclipse Che Custom Resource
+  OAUTH_CLIENT_NAME=$(oc get checluster eclipse-che -n eclipse-che -o=jsonpath='{.spec.auth.oAuthClientName}')
+
+  # Obtain from Keycloak the token to make api request authentication
+  IDP_TOKEN=$(curl -k --location --request POST ''$IDP_HOST'/auth/realms/master/protocol/openid-connect/token' \
+  --header 'Content-Type: application/x-www-form-urlencoded' \
+  --data-urlencode 'username=admin' \
```
Comment (tolusha) [inline .github/bin/oauth-provision.sh:41]: Can we put this  `admin` in a varieable ?
CHE_USERNAME="admin"

**Review by tolusha — APPROVED** [review]

File: .github/bin/oauth-provision.sh [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    IDP_HOST="https://"$(oc get route keycloak -n eclipse-che -o=jsonpath='{.spec.host}')
+  fi
+
+  # Get the oauth client from Eclipse Che Custom Resource
+  OAUTH_CLIENT_NAME=$(oc get checluster eclipse-che -n eclipse-che -o=jsonpath='{.spec.auth.oAuthClientName}')
+
+  # Obtain from Keycloak the token to make api request authentication
+  IDP_TOKEN=$(curl -k --location --request POST ''$IDP_HOST'/auth/realms/master/protocol/openid-connect/token' \
+  --header 'Content-Type: application/x-www-form-urlencoded' \
+  --data-urlencode 'username='$CHE_USERNAME'' \
```
Comment (tolusha) [inline .github/bin/oauth-provision.sh:41]: IDP_USER

File: .github/bin/oauth-provision.sh [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  --data-urlencode 'password='$IDP_PASSWORD'' \
+  --data-urlencode 'grant_type=password' \
+  --data-urlencode 'client_id=admin-cli' | jq -r .access_token)
+
+  echo -e "[INFO] IDP Token: $IDP_TOKEN"
+
+  # Get admin user id from IDP
+  CHE_USER_ID=$(curl --location -k --request GET ''$IDP_HOST'/auth/admin/realms/che/users' \
+  --header 'Authorization: Bearer '$IDP_TOKEN'' | jq -r '.[] | select(.username == "admin").id' )
```
Comment (tolusha) [inline .github/bin/oauth-provision.sh:50]: CHE_USERNAME

File: .github/bin/oauth-provision.sh [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+userUID: $OCP_USER_UID
+scopes:
+  - 'user:full'
+EOF
+)
+  # Create SQL script
+  echo -e "Created authorization client: $OAUTHCLIENTAuthorization"
+  cat << 'EOF' > path.sql
+UPDATE federated_identity SET token ='{"access_token":"INSERT_TOKEN_HERE","expires_in":86400,"scope":"user:full","token_type":"Bearer"}'
+WHERE federated_username = 'admin'
```
Comment (tolusha) [inline .github/bin/oauth-provision.sh:83]: CHE_USERNAME

**Review by tolusha — APPROVED** [review]

**Pull request merged**